pkg/operator/options/options.go

```diff
@@ -57,113 +60,95 @@ type OptionFields struct {
-	f.BoolVar(&opts.EnableProfiling, "enable-profiling", env.WithDefaultBool("ENABLE_PROFILING", false), "Enable the profiling on the metric endpoint")
-	f.BoolVar(&opts.EnableLeaderElection, "leader-elect", env.WithDefaultBool("LEADER_ELECT", true), "Start leader election client and gain leadership before executing the main loop. Enable this when running replicated components for high availability.")
-	f.Int64Var(&opts.MemoryLimit, "memory-limit", env.WithDefaultInt64("MEMORY_LIMIT", -1), "Memory limit on the container running the controller. The GC soft memory limit is set to 90% of this value.")
-	f.StringVar(&opts.LogLevel, "log-level", env.WithDefaultString("LOG_LEVEL", ""), "Log verbosity level. Can be one of 'debug', 'info', or 'error'")
-
-	// Vars that must be merged with settings
-	f.DurationVar(&opts.BatchMaxDuration, "batch-max-duration", env.WithDefaultDuration("BATCH_MAX_DURATION", 10*time.Second), "The maximum length of a batch window. The longer this is, the more pods we can consider for provisioning at one time which usually results in fewer but larger nodes.")
-	f.DurationVar(&opts.BatchIdleDuration, "batch-idle-duration", env.WithDefaultDuration("BATCH_IDLE_DURATION", time.Second), "The maximum amount of time with no new pending pods that if exceeded ends the current batching window. If pods arrive faster than this time, the batching window will be extended up to the maxDuration. If they arrive slower, the pods will be batched separately.")
-	f.StringVar(&opts.FeatureGates.inputStr, "feature-gates", env.WithDefaultString("FEATURE_GATES", "Drift=false"), "Optional features can be enabled / disabled using feature gates. Current options are: Drift")
-
-	return opts
-}
-
-func (o *Options) Parse(args ...string) (*Options, error) {
-	if err := o.FlagSet.Parse(args); err != nil {
+func (o *Options) Parse(fs *flag.FlagSet, args ...string) error {
+	if err := fs.Parse(args); err != nil {
 		if errors.Is(err, flag.ErrHelp) {
 			os.Exit(0)
 		}
-		return nil, err
+		return err
 	}
 
 	if !lo.Contains(validLogLevels, o.LogLevel) {
-		return nil, fmt.Errorf("failed to validate cli flags / env vars, invalid log level %q", o.LogLevel)
-	}
-
-	o.FeatureGates = MustParseFeatureGates(o.FeatureGates.inputStr)
-
-	// Check if shared fields have been set. If they haven't, they may be ovewritten by settings parsed from configmaps.
-	o.FlagSet.Visit(func(f *flag.Flag) {
-		switch f.Name {
-		case "batch-max-duration":
-			o.BatchMaxDurationSet = true
-		case "batch-idle-duration":
-			o.BatchIdleDurationSet = true
-		case "feature-gates":
-			o.FeatureGatesSet = true
-		}
-	})
-	if _, ok := os.LookupEnv("BATCH_MAX_DURATION"); ok {
-		o.BatchMaxDurationSet = true
-	}
-	if _, ok := os.LookupEnv("BATCH_IDLE_DURATION"); ok {
-		o.BatchIdleDurationSet = true
+		return fmt.Errorf("failed to validate cli flags / env vars, invalid log level %q", o.LogLevel)
```

Comment:
```suggestion
		return fmt.Errorf("validating cli flags / env vars, invalid log level %q", o.LogLevel)
```

nit: style with wording